pkg/querysharding/analyzer_test.go

```diff
@@ -165,7 +167,10 @@ http_requests_total`,
 
 	for _, test := range nonShardable {
 		t.Run(test.name, func(t *testing.T) {
-			analyzer := NewQueryAnalyzer()
+			analyzer, err := NewQueryAnalyzer()
+			if err != nil {
+				fmt.Println("error intializing cache: ", err)
+			}
```

Comment:
Let's not check error and use `require.NoError(t, err)` instead.

Reply:
ok